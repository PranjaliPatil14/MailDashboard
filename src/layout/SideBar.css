.sidebar {
  width: 80px;
  font-size: 16px;
  height: 100%;
  background: #2f4050;
  color: #a7b1c2;
  display: flex;
  flex-direction: column;
  overflow: auto;
}

.sidebar.expanded {
  width: 250px;
  align-items: flex-start;
}

.sidebar--menu {
  width: 100%;
  font-weight: bold;
  display: flex;
  align-items: center;
  flex: 1;
  flex-direction: column;
  list-style: none;
}

.sidebar--menu.expanded {
  align-items: unset;
}

.sidebar--menu-item {
  margin: 15px 0;
}


.sidebar--menu-item.expanded{
  width: 100%;
  display: flex;
  align-items: center;
  margin: 15px 0;
}

.sidebar--toggle {
  width: 100%;
  outline: none;
  box-shadow: none;
  border: none;
  color: white;
  background: #243846;
}

.menu-icon {
  display: flex;
  flex: 0.2;
  justify-self: flex-end;
}

.menu-title {
  flex: 0.6;
  display: flex;
  justify-self: flex-start;
}

.menu-toggle {
  flex: 0.2;
  display: flex;
  justify-self: flex-start;
}

.active-menu {
  background: #293846;
  padding: 10px 0;
  border-left: 5px solid #05AB8C;
  color: white;
}

.submenu {
  display: flex;
  flex: 1;
  flex-direction: column;
  list-style: none;
  padding: 0 50px;
  color: #A7B1C2;
}

.submenu-item {
  margin: 10px 0;
}

.submenu-item:first-child{
  color: white;
}
.user-profile {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  padding: 40px 0 20px 20px;
}

.user-profile--name {
  font-size: 16px;
  color: #D1D6E0;
}

.user-profile--role {
  font-size: 14px;
  color: #7F95A9;
}

.user-profile--role>span {
  font-size: 8px;
}

.user-profile--image {
  height: 55px;
  width: 60px;
  margin-bottom: 10px;
  border-radius: 50%;
}
