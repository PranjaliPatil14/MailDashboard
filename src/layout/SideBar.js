import React, { useState } from "react";
import { ChevronLeft, ChevronDown } from "react-feather";
import ProfileImage from "../assets/images/profile-image.png";

import "./SideBar.css";
import menuConfig from "./menuConfig";

const SideBar = () => {
  const [expanded, setExpanded] = useState(false);

  const getIcons = () => {
    return (
      <>
        {menuConfig.map(({ Component, key }) => (
          <li key={key} className="sidebar--menu-item">
            <Component className="menu-icon" />
          </li>
        ))}
      </>
    );
  };

  const getToggleIcon = (active, submenu) => {
    if (active && submenu.length) {
      return <ChevronDown />;
    }
    if (submenu.length) {
      return <ChevronLeft />;
    }
    return "";
  };

  const getMenu = () => {
    return (
      <>
        {menuConfig.map(({ Component, key, title, subMenu, active }) => (
          <div className={active ? "active-menu" : ""} key={key}>
            <li className="sidebar--menu-item expanded">
              <Component className="menu-icon" />
              <span className="menu-title">{title}</span>
              <span className="menu-toggle">
                {getToggleIcon(active, subMenu)}
              </span>
            </li>

            <ul className="submenu">
              {subMenu.length && active ? (
                subMenu.map(({ title: menuTitle }) => (
                  <li key={menuTitle} className="submenu-item">
                    <span className="menu-title">{menuTitle}</span>
                  </li>
                ))
              ) : (
                <></>
              )}
            </ul>
          </div>
        ))}
      </>
    );
  };

  // TODO: toggle sidebar on outside click

  return (
    <nav className={`sidebar ${expanded ? "expanded" : ""}`}>
      <button
        className="sidebar--toggle"
        onClick={() => setExpanded(!expanded)}
        type="submit"
      >
        {expanded ? (
          <div className="user-profile">
            <img
              src={ProfileImage}
              alt="profile"
              className="user-profile--image"
            />
            <strong className="user-profile--name">User Name</strong>
            <p className="user-profile--role">
              role <span>â–¼</span>
            </p>
          </div>
        ) : (
          <h2 className="sidebar--toggle-text">IN+</h2>
        )}
      </button>
      <ul className={`sidebar--menu ${expanded ? "expanded" : ""}`}>
        {expanded ? getMenu() : getIcons()}
      </ul>
    </nav>
  );
};

export default SideBar;
