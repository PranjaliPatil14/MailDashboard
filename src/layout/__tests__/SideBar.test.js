import React from "react";
import { render, fireEvent } from "@testing-library/react";
import SideBar from "../SideBar";

describe("SideBar", () => {
  test("should render sidebar", () => {
    const { asFragment } = render(<SideBar />);

    expect(asFragment()).toMatchSnapshot();
  });

  test("should render expanded sidebar", () => {
    const { asFragment, getByRole } = render(<SideBar />);

    fireEvent.click(getByRole("button"));

    expect(asFragment()).toMatchSnapshot();
  });
});
