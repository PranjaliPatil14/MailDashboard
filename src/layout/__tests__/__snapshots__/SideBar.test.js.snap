// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar should render expanded sidebar 1`] = `
<DocumentFragment>
  <nav
    class="sidebar expanded"
  >
    <button
      class="sidebar--toggle"
      type="submit"
    >
      <div
        class="user-profile"
      >
        <img
          alt="profile"
          class="user-profile--image"
          src="profile-image.png"
        />
        <strong
          class="user-profile--name"
        >
          User Name
        </strong>
        <p
          class="user-profile--role"
        >
          role
        </p>
      </div>
    </button>
    <ul
      class="sidebar--menu expanded"
    >
      <div
        class=""
      >
        <li
          class="sidebar--menu-item expanded"
        >
          <svg
            class="menu-icon"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="7"
              width="7"
              x="3"
              y="3"
            />
            <rect
              height="7"
              width="7"
              x="14"
              y="3"
            />
            <rect
              height="7"
              width="7"
              x="14"
              y="14"
            />
            <rect
              height="7"
              width="7"
              x="3"
              y="14"
            />
          </svg>
          <span
            class="menu-title"
          >
            Dashboards
          </span>
          <span
            class="menu-toggle"
          >
            <svg
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="15 18 9 12 15 6"
              />
            </svg>
          </span>
        </li>
        <ul
          class="submenu"
        />
      </div>
      <div
        class=""
      >
        <li
          class="sidebar--menu-item expanded"
        >
          <svg
            class="menu-icon"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="8"
              r="7"
            />
            <polyline
              points="8.21 13.89 7 23 12 20 17 23 15.79 13.88"
            />
          </svg>
          <span
            class="menu-title"
          >
            Layouts
          </span>
          <span
            class="menu-toggle"
          />
        </li>
        <ul
          class="submenu"
        />
      </div>
      <div
        class=""
      >
        <li
          class="sidebar--menu-item expanded"
        >
          <svg
            class="menu-icon"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="18"
              x2="18"
              y1="20"
              y2="10"
            />
            <line
              x1="12"
              x2="12"
              y1="20"
              y2="4"
            />
            <line
              x1="6"
              x2="6"
              y1="20"
              y2="14"
            />
          </svg>
          <span
            class="menu-title"
          >
            Graphs
          </span>
          <span
            class="menu-toggle"
          />
        </li>
        <ul
          class="submenu"
        />
      </div>
      <div
        class="active-menu"
      >
        <li
          class="sidebar--menu-item expanded"
        >
          <svg
            class="menu-icon"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4 4h16c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H4c-1.1 0-2-.9-2-2V6c0-1.1.9-2 2-2z"
            />
            <polyline
              points="22,6 12,13 2,6"
            />
          </svg>
          <span
            class="menu-title"
          >
            Mailbox
          </span>
          <span
            class="menu-toggle"
          >
            <svg
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="6 9 12 15 18 9"
              />
            </svg>
          </span>
        </li>
        <ul
          class="submenu"
        >
          <li
            class="submenu-item"
          >
            <span
              class="menu-title"
            >
              Inbox
            </span>
          </li>
          <li
            class="submenu-item"
          >
            <span
              class="menu-title"
            >
              Email view
            </span>
          </li>
          <li
            class="submenu-item"
          >
            <span
              class="menu-title"
            >
              Compose email
            </span>
          </li>
          <li
            class="submenu-item"
          >
            <span
              class="menu-title"
            >
              Email templates
            </span>
          </li>
        </ul>
      </div>
      <div
        class=""
      >
        <li
          class="sidebar--menu-item expanded"
        >
          <svg
            class="menu-icon"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M21.21 15.89A10 10 0 1 1 8 2.83"
            />
            <path
              d="M22 12A10 10 0 0 0 12 2v10z"
            />
          </svg>
          <span
            class="menu-title"
          >
            Metrics
          </span>
          <span
            class="menu-toggle"
          />
        </li>
        <ul
          class="submenu"
        />
      </div>
      <div
        class=""
      >
        <li
          class="sidebar--menu-item expanded"
        >
          <svg
            class="menu-icon"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19 16.9A5 5 0 0 0 18 7h-1.26a8 8 0 1 0-11.62 9"
            />
            <polyline
              points="13 11 9 17 15 17 11 23"
            />
          </svg>
          <span
            class="menu-title"
          >
            Widgets
          </span>
          <span
            class="menu-toggle"
          />
        </li>
        <ul
          class="submenu"
        />
      </div>
      <div
        class=""
      >
        <li
          class="sidebar--menu-item expanded"
        >
          <svg
            class="menu-icon"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11 4H4a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
            />
            <path
              d="M18.5 2.5a2.121 2.121 0 0 1 3 3L12 15l-4 1 1-4 9.5-9.5z"
            />
          </svg>
          <span
            class="menu-title"
          >
            Forms
          </span>
          <span
            class="menu-toggle"
          >
            <svg
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="15 18 9 12 15 6"
              />
            </svg>
          </span>
        </li>
        <ul
          class="submenu"
        />
      </div>
      <div
        class=""
      >
        <li
          class="sidebar--menu-item expanded"
        >
          <svg
            class="menu-icon"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="14"
              rx="2"
              ry="2"
              width="20"
              x="2"
              y="3"
            />
            <line
              x1="8"
              x2="16"
              y1="21"
              y2="21"
            />
            <line
              x1="12"
              x2="12"
              y1="17"
              y2="21"
            />
          </svg>
          <span
            class="menu-title"
          >
            App Views
          </span>
          <span
            class="menu-toggle"
          />
        </li>
        <ul
          class="submenu"
        />
      </div>
      <div
        class=""
      >
        <li
          class="sidebar--menu-item expanded"
        >
          <svg
            class="menu-icon"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="13"
              rx="2"
              ry="2"
              width="13"
              x="9"
              y="9"
            />
            <path
              d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
            />
          </svg>
          <span
            class="menu-title"
          >
            Files
          </span>
          <span
            class="menu-toggle"
          />
        </li>
        <ul
          class="submenu"
        />
      </div>
      <div
        class=""
      >
        <li
          class="sidebar--menu-item expanded"
        >
          <svg
            class="menu-icon"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <line
              x1="2"
              x2="22"
              y1="12"
              y2="12"
            />
            <path
              d="M12 2a15.3 15.3 0 0 1 4 10 15.3 15.3 0 0 1-4 10 15.3 15.3 0 0 1-4-10 15.3 15.3 0 0 1 4-10z"
            />
          </svg>
          <span
            class="menu-title"
          >
            World
          </span>
          <span
            class="menu-toggle"
          />
        </li>
        <ul
          class="submenu"
        />
      </div>
      <div
        class=""
      >
        <li
          class="sidebar--menu-item expanded"
        >
          <svg
            class="menu-icon"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10.29 3.86L1.82 18a2 2 0 0 0 1.71 3h16.94a2 2 0 0 0 1.71-3L13.71 3.86a2 2 0 0 0-3.42 0z"
            />
            <line
              x1="12"
              x2="12"
              y1="9"
              y2="13"
            />
            <line
              x1="12"
              x2="12.01"
              y1="17"
              y2="17"
            />
          </svg>
          <span
            class="menu-title"
          >
            Alerts
          </span>
          <span
            class="menu-toggle"
          />
        </li>
        <ul
          class="submenu"
        />
      </div>
      <div
        class=""
      >
        <li
          class="sidebar--menu-item expanded"
        >
          <svg
            class="menu-icon"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="15"
              rx="2"
              ry="2"
              width="20"
              x="2"
              y="7"
            />
            <polyline
              points="17 2 12 7 7 2"
            />
          </svg>
          <span
            class="menu-title"
          >
            Monitor
          </span>
          <span
            class="menu-toggle"
          />
        </li>
        <ul
          class="submenu"
        />
      </div>
    </ul>
  </nav>
</DocumentFragment>
`;

exports[`SideBar should render sidebar 1`] = `
<DocumentFragment>
  <nav
    class="sidebar "
  >
    <button
      class="sidebar--toggle"
      type="submit"
    >
      <h2
        class="sidebar--toggle-text"
      >
        IN+
      </h2>
    </button>
    <ul
      class="sidebar--menu "
    >
      <li
        class="sidebar--menu-item"
      >
        <svg
          class="menu-icon"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            height="7"
            width="7"
            x="3"
            y="3"
          />
          <rect
            height="7"
            width="7"
            x="14"
            y="3"
          />
          <rect
            height="7"
            width="7"
            x="14"
            y="14"
          />
          <rect
            height="7"
            width="7"
            x="3"
            y="14"
          />
        </svg>
      </li>
      <li
        class="sidebar--menu-item"
      >
        <svg
          class="menu-icon"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="8"
            r="7"
          />
          <polyline
            points="8.21 13.89 7 23 12 20 17 23 15.79 13.88"
          />
        </svg>
      </li>
      <li
        class="sidebar--menu-item"
      >
        <svg
          class="menu-icon"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="18"
            x2="18"
            y1="20"
            y2="10"
          />
          <line
            x1="12"
            x2="12"
            y1="20"
            y2="4"
          />
          <line
            x1="6"
            x2="6"
            y1="20"
            y2="14"
          />
        </svg>
      </li>
      <li
        class="sidebar--menu-item"
      >
        <svg
          class="menu-icon"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4 4h16c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H4c-1.1 0-2-.9-2-2V6c0-1.1.9-2 2-2z"
          />
          <polyline
            points="22,6 12,13 2,6"
          />
        </svg>
      </li>
      <li
        class="sidebar--menu-item"
      >
        <svg
          class="menu-icon"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21.21 15.89A10 10 0 1 1 8 2.83"
          />
          <path
            d="M22 12A10 10 0 0 0 12 2v10z"
          />
        </svg>
      </li>
      <li
        class="sidebar--menu-item"
      >
        <svg
          class="menu-icon"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 16.9A5 5 0 0 0 18 7h-1.26a8 8 0 1 0-11.62 9"
          />
          <polyline
            points="13 11 9 17 15 17 11 23"
          />
        </svg>
      </li>
      <li
        class="sidebar--menu-item"
      >
        <svg
          class="menu-icon"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11 4H4a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
          />
          <path
            d="M18.5 2.5a2.121 2.121 0 0 1 3 3L12 15l-4 1 1-4 9.5-9.5z"
          />
        </svg>
      </li>
      <li
        class="sidebar--menu-item"
      >
        <svg
          class="menu-icon"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            height="14"
            rx="2"
            ry="2"
            width="20"
            x="2"
            y="3"
          />
          <line
            x1="8"
            x2="16"
            y1="21"
            y2="21"
          />
          <line
            x1="12"
            x2="12"
            y1="17"
            y2="21"
          />
        </svg>
      </li>
      <li
        class="sidebar--menu-item"
      >
        <svg
          class="menu-icon"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            height="13"
            rx="2"
            ry="2"
            width="13"
            x="9"
            y="9"
          />
          <path
            d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
          />
        </svg>
      </li>
      <li
        class="sidebar--menu-item"
      >
        <svg
          class="menu-icon"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <line
            x1="2"
            x2="22"
            y1="12"
            y2="12"
          />
          <path
            d="M12 2a15.3 15.3 0 0 1 4 10 15.3 15.3 0 0 1-4 10 15.3 15.3 0 0 1-4-10 15.3 15.3 0 0 1 4-10z"
          />
        </svg>
      </li>
      <li
        class="sidebar--menu-item"
      >
        <svg
          class="menu-icon"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10.29 3.86L1.82 18a2 2 0 0 0 1.71 3h16.94a2 2 0 0 0 1.71-3L13.71 3.86a2 2 0 0 0-3.42 0z"
          />
          <line
            x1="12"
            x2="12"
            y1="9"
            y2="13"
          />
          <line
            x1="12"
            x2="12.01"
            y1="17"
            y2="17"
          />
        </svg>
      </li>
      <li
        class="sidebar--menu-item"
      >
        <svg
          class="menu-icon"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            height="15"
            rx="2"
            ry="2"
            width="20"
            x="2"
            y="7"
          />
          <polyline
            points="17 2 12 7 7 2"
          />
        </svg>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
